spyderlib/plugins/editor.py

```diff
@@ -492,7 +478,49 @@ def __init__(self, parent, ignore_last_opened_files=False):
             self.add_cursor_position_to_history(filename, position)
         self.update_cursorpos_actions()
         self.set_path()
-        
+
+    def set_open_filenames(self):
+        """ """
+        editorstack = self.editorstacks[0]
+        if self.projectexplorer:
+            if not self.projectexplorer.get_active_project():
+                filenames = []
+                filenames += [finfo.filename for finfo in editorstack.data]
+                self.set_option('filenames', filenames)
+
+    def setup_open_files(self):
+        """ """
+        active_project_path = None
+        if self.projectexplorer:
+             active_project_path = self.projectexplorer.get_active_project_path()
+
+        spyder_filenames = self.get_option('filenames', [])
+        project_filenames = self.projectexplorer.get_project_filenames()
+        filenames = spyder_filenames[:]

```

Comment:
Basically the editor is in charge of opening the files, so depending if there was an active project opened, those files should be open, otherwise open the normal files (stored in the spyder.ini)

Do you have a different idea how it should be done?